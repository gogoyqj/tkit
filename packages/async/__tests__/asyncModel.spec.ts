/**
 * @description 确保接口类型正常
 */
import { parser } from './parser';

const files = 'asyncFail.tsx asyncOK.tsx doFail.ts doOK.ts'.split(' ');

describe('tkit-async work ok', () => {
  const errors = parser(files, 'samples');
  files.forEach(file =>
    it(`check sample/${file} work ok`, () => {
      const current = errors[file];
      if (file.match(/OK/)) {
        expect(current).toHaveLength(0);
      } else if (file.match(/Fail/)) {
        expect(current.length).toBeGreaterThan(0);
      }
      expect(current).toMatchSnapshot();
    })
  );
});
