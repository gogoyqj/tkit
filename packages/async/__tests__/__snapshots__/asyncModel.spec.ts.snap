// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tkit-async work ok check sample/asyncFail.tsx work ok 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/asyncFail.tsx",
    "messageText": "Type '(props: { wrong: number; }) => any' is not assignable to type 'ComponentType<AsyncFormProps>'.
  Type '(props: { wrong: number; }) => any' is not assignable to type 'FunctionComponent<AsyncFormProps>'.
    Types of parameters 'props' and 'props' are incompatible.
      Property 'wrong' is missing in type 'PropsWithChildren<AsyncFormProps>' but required in type '{ wrong: number; }'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/asyncFail.tsx",
    "messageText": "Type '(props: { wrong: number; }) => any' is not assignable to type 'ComponentType<{ status: AsyncStatus; }>'.
  Type '(props: { wrong: number; }) => any' is not assignable to type 'FunctionComponent<{ status: AsyncStatus; }>'.
    Types of parameters 'props' and 'props' are incompatible.
      Property 'wrong' is missing in type 'PropsWithChildren<{ status: AsyncStatus; }>' but required in type '{ wrong: number; }'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/asyncFail.tsx",
    "messageText": "Type '(props: { wrong: number; }) => any' is not assignable to type '(e: AsyncResultEventType) => any'.
  Types of parameters 'props' and 'e' are incompatible.
    Property 'wrong' is missing in type 'AsyncResultEventType' but required in type '{ wrong: number; }'.",
  },
]
`;

exports[`tkit-async work ok check sample/asyncOK.tsx work ok 1`] = `Array []`;

exports[`tkit-async work ok check sample/doFail.ts work ok 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/doFail.ts",
    "messageText": "Type 'string' is not assignable to type 'number'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/doFail.ts",
    "messageText": "Type 'string' is not assignable to type '[number]'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/doFail.ts",
    "messageText": "Type 'string' is not assignable to type 'never'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/doFail.ts",
    "messageText": "Type 'string' is not assignable to type '[number, string]'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/doFail.ts",
    "messageText": "Type 'string' is not assignable to type 'number'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/doFail.ts",
    "messageText": "Type 'string' is not assignable to type '[number]'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/doFail.ts",
    "messageText": "Type 'string' is not assignable to type 'never'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "samples/doFail.ts",
    "messageText": "Type 'string' is not assignable to type '[number, string]'.",
  },
]
`;

exports[`tkit-async work ok check sample/doOK.ts work ok 1`] = `Array []`;
